### PR TITLE
WIP: OSX toolchain bump

### PR DESCRIPTION
**Pull request by theuni**
This bumps us to:
- SDK: 10.11
- clang: 3.7.1
- cctools: 877.8
- ld64: 253.9

Beyond just keeping us current, this is also necessary for #7770.
The bump doesn't change anything in terms of back-compat, we still specify 10.7 as the required minimum.

Todo before merge:
- [ ] Depends on #8053. The qt change here should be reverted after that.
- [ ] I need to thoroughly review the cctools/ld64 bump, this was just a quick port.
- [ ] Docs need to be updated to reflect the switch to 10.11 sdk.

### CODE REVIEW

File: depends/packages/qt.mk [hunk 1]
```diff
@@ -124,6 +124,7 @@ define $(package)_preprocess_cmds
   sed -i.old "s/src_plugins.depends = src_sql src_xml src_network/src_plugins.depends = src_xml src_network/" qtbase/src/src.pro && \
   sed -i.old "s|X11/extensions/XIproto.h|X11/X.h|" qtbase/src/plugins/platforms/xcb/qxcbxsettings.cpp && \
   sed -i.old 's/if \[ "$$$$XPLATFORM_MAC" = "yes" \]; then xspecvals=$$$$(macSDKify/if \[ "$$$$BUILD_ON_MAC" = "yes" \]; then xspecvals=$$$$(macSDKify/' qtbase/configure && \
+  sed -i.old 's/CGEventCreateMouseEvent(0, kCGEventMouseMoved, pos, 0)/CGEventCreateMouseEvent(0, kCGEventMouseMoved, pos, kCGMouseButtonLeft)/' qtbase/src/plugins/platforms/cocoa/qcocoacursor.mm && \
```
Comment (jonasschnelli) [inline depends/packages/qt.mk]: hmm... what's the patch for? IMO the `kCGMouseButtonLeft ` constant is set to 0 as well? https://developer.apple.com/library/mac/documentation/Carbon/Reference/QuartzEventServicesRef/index.html#//apple_ref/c/econst/kCGMouseButtonLeft

Comment (theuni) [inline depends/packages/qt.mk]: See the todo in description, this was fixed in 5.6.

You'd have to cast like (CGMouseButton)0.

**Comment by theuni** [comment]
@jonasschnelli afaik 10.7 is the pretty standard minimum these days, but we could always bump if desired. It doesn't look like that runtime check really hurts anything, though?

**Comment by fanquake** [comment]
@jonasschnelli Is that the only 10.8 related hack in the codebase?

**Comment by jonasschnelli** [comment]
@fanquake: bumping to 10.8 could allow use to use the NSNotificationCenter without weak linking.

But after thinking about it, I guess we should keep 10.7 as min for 0.13 (maybe also for 0.14).

**Comment by jonasschnelli** [comment]
Tested ACK f5ade22e2200d729a2cec1e95a5aaa98fcb41364.

Did a gitian build, ran binaries on 10.11 and on a 10.07 machine:
bildschirmfoto 2016-06-10 um 11 42 04

**Comment by laanwj** [comment]
> Question: would it be to restrictive to set 10.8 as minimum deployment version (OSX_MIN_VERSION)? It would allow us to get rid of some hack in

I think that's a pretty weak reason to drop support for an OS version. I'd also say stick with 10.7 as minimum for the foreseeable future.

**Comment by fanquake** [comment]
I've put together some changes to the build-notes [here](https://github.com/fanquake/bitcoin/commit/4f44073d6db9c9b462979bfba1103f19e7a6ad7e) if you'd like to cherry pick.

**Comment by laanwj** [comment]
Closing in favor of #8210

@fanquake Probably want to PR those separately

**Pull request closed without merging**